### ISSUE TITLE
False positive when TypeVar default references another constrained TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7446]
PEP 696 indicates that the following should type check without errors:

```python
Y2 = TypeVar("Y2", int, str)
TypeVar("AlsoOk2", int, str, bool, default=Y2)  # OK
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.354, which I just published.
